python_modules/libraries/dagster-dg/dagster_dg/config.py

```diff
@@ -390,11 +402,54 @@ def is_project_file_config(config: "DgFileConfig") -> TypeGuard[DgProjectFileCon
 DgFileConfig: TypeAlias = Union[DgWorkspaceFileConfig, DgProjectFileConfig]
 
 
+def is_dg_specific_config_file(path: Path) -> bool:
+    """Check if the file is a dg-specific toml file."""
+    return path.name == "dg.toml" or path.name == ".dg.toml"
+
+
+@contextmanager
+def modify_dg_toml_config(path: Path) -> Iterator[Union[tomlkit.TOMLDocument, tomlkit.items.Table]]:
+    """Modify a TOML file as a tomlkit.TOMLDocument, preserving comments and formatting."""
+    with modify_toml(path) as toml:
+        if is_dg_specific_config_file(path):
+            yield toml
+        elif not has_toml_node(toml, ("tool", "dg")):
+            raise KeyError(
+                "TOML file does not have a tool.dg section. This is required for pyproject.toml files."
+            )
+        else:
+            yield get_toml_node(toml, ("tool", "dg"), tomlkit.items.Table)
+
+
+@contextmanager
+def modify_dg_toml_config_as_dict(path: Path) -> Iterator[dict[str, Any]]:
```

Comment:
Is this only used in test or something? This seems dangerous to do for any prod codepaths

Reply:
>Is this only used in test or something?

Yes, I'll add a comment